### ISSUE TITLE
No 'monthview-primary-with-event' class added when assigning a new array programatically.

### ISSUE THREAD

**Issue by milk-shake** [twinssbc/Ionic2-Calendar#113]
I'm getting dates from an API and mapping those to new event objects and assigning that to the eventSource property.

The events seem to be assigned, clicking a date to remove from the array works as expected but the class is not added to the element.

**Comment by milk-shake** [comment]
found an issue here:
Line 549 in monthview.ts
```
while (index < timeDifferenceEnd - eps) {
                dates[index].hasEvent = true;
                let eventSet = dates[index].events;
                if (eventSet) {
                    eventSet.push(event);
                } else {
                    eventSet = [];
                    eventSet.push(event);
                    dates[index].events = eventSet;
                }
                index += 1;
            }
```

```index``` is _never_ lower than ```timeDifferenceEnd - eps```

I'm not totally sure what ```eps``` is, but for my use case it's simpler to replace that code with
```
dates[index].hasEvent = true;
dates[index].events = dates[index].events ? dates[index].events : [];
dates[index].events.push(event);
```

because there will only ever be one event.

**Comment by twinssbc** [comment]
@milk-shake I'm not quite sure what issue do you want to solve here.
You are saying 
> The events seem to be assigned,
Do you mean actually you didn't see the cell background turns to blue?
What if you click this issue, are the events happen on that day displaying in the below event details list?

Regarding your second comment, this piece of code is to calculate a particular should occupy how many days and set the flag accordingly. Since the calculation precision of javascript is not too accurate, so I need to add eps to tolerate some deviation.

**Comment by milk-shake** [comment]
The cell did not change to blue, and under further investigation the event was not assigned to the date either.

Clicking the days activated an event on the day, but assigning an array of event objects did not due to something at line 549 meaning the while loop never executing.

**Comment by twinssbc** [comment]
Usually when user reports the issue that they have assigned the eventSource, but not displayed in the calendar, the root cause is either the assignment doesn't happen in the async callback or the array element is manipulated directly.

**Comment by milk-shake** [comment]
I checked the other issues and saw that was common, but I am definitely re-assigning a clone because I'm filtering through and removing array elements on every onTimeSelected call and assigning the returned array to ```eventSource```. 

I'm also definitely inside my subscription callback because I have the dates and they loop through. 

Like I said, I've forked the code and fixed it for myself, but there seems to be an issue when assigning events programatically and not through click events.

Example:
```
let currentUnavailableDays = this.unavailableDays.filter(
      (item) => {
        if(new Date(item.startTime).setHours(0,0,0,0) != new Date(eventDate).setHours(0,0,0,0)) {
          return item;
        }
    });
```

I then filter out undefined/null values and then:
```
this.unavailableDays = currentUnavailableDays;
```

where this.unavailableDays is assigned to the ```eventSource``` property.

This function is called on API success and when a date is clicked (via onTimeSelected event binding)
Events aren't attached when I call it from the subscription call back, but are attached when I click on the date - same function call, same data.

when I logged out: 
```
index < timeDifferenceEnd - eps
```

while debugging, index was never below ```timeDifferenceEnd - eps```, so the event never got assigned to the date.
